File: src/main/java/com/nugurang/graphql/mutation/ArticleMutation.java
package com.nugurang.graphql.mutation;

import com.nugurang.dto.ArticleDto;
import com.nugurang.dto.ArticleInputDto;
import com.nugurang.service.ArticleService;
import graphql.kickstart.tools.GraphQLMutationResolver;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class ArticleMutation implements GraphQLMutationResolver {
    private final ArticleService articleService;

    public ArticleDto createArticle(ArticleInputDto articleInputDto, Long thread, Optional<Long> parent) {
        return articleService.createArticle(articleInputDto, thread, parent).toDto();
    }

    public ArticleDto updateArticle(ArticleInputDto articleInputDto, Long id) {
        return articleService.updateArticle(articleInputDto, id).toDto();
    }

    public Long deleteArticle(Long id) {
        articleService.deleteArticle(id);
        return id;
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public ArticleMutation(final ArticleService articleService) {
        this.articleService = articleService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/mutation/ArticleMutation.kt
package com.nugurang.graphql.mutation

import com.nugurang.dto.ArticleDto
import com.nugurang.dto.ArticleInputDto
import com.nugurang.mapper.ArticleMapper
import com.nugurang.service.ArticleService
import graphql.kickstart.tools.GraphQLMutationResolver
import org.springframework.stereotype.Service

@Service
class ArticleMutation(
    private val articleService: ArticleService,
    private val articleMapper: ArticleMapper
) : GraphQLMutationResolver {
    fun createArticle(articleInputDto: ArticleInputDto, thread: Long, parent: Long?): ArticleDto {
        return articleService.createArticle(articleInputDto, thread, parent).let(articleMapper::toDto)
    }

    fun updateArticle(articleInputDto: ArticleInputDto, id: Long): ArticleDto {
        return articleService.updateArticle(articleInputDto, id).let(articleMapper::toDto)
    }

    fun deleteArticle(id: Long): Long {
        articleService.deleteArticle(id)
        return id
    }
}